### PR TITLE
Report for pilot screen

### PR DESCRIPTION
**Pull request by Baribal02**
The form screens to fill the crew and pilot reports.

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/screens/reports/CrewReportScreenTest.kt [hunk 1]
```diff
@@ -1,29 +1,118 @@
...
+      navController.navigate(Route.CREW_REPORT)
+    }
+  }
+
+  @Test
+  fun allFieldsAreDisplayed() {
+    composeTestRule
+        .onNodeWithTag("Crew Report LazyColumn")
+        .performScrollToNode(hasTestTag("Number of little champagne bottles"))
+    composeTestRule.onNodeWithTag("Number of little champagne bottles").assertExists()
```
Comment (Prolas) [inline app/src/androidTest/java/ch/epfl/skysync/screens/reports/CrewReportScreenTest.kt:49]: Is it really useful to check if the node exists if it managed to scroll to it ?

Comment (Baribal02) [reply app/src/androidTest/java/ch/epfl/skysync/screens/reports/CrewReportScreenTest.kt:49]: Not really, but, in my opinion, it makes clearer what the test is actually testing on this component

File: app/src/main/java/ch/epfl/skysync/util/FormsUtil.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package ch.epfl.skysync.util
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Date
+
+/** Useful methods for input validation */
+fun inputValidation(vararg errors: Boolean): Boolean {
```
Comment (mathieumognol) [inline app/src/main/java/ch/epfl/skysync/util/FormsUtil.kt:10]: Name could be misleading, would rename it `hasAnyError`

**Review by mathieumognol — APPROVED** [review]
LGTM

File: app/src/main/java/ch/epfl/skysync/components/forms/BaseReportForm.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    onConfirm: (Vehicle, String) -> Unit,
+    vehicleProblem: MutableMap<Vehicle, String>,
+    comments: String,
+    onBeginTimeChange: (Date) -> Unit,
+    onEndTimeChange: (Date) -> Unit,
+    onPauseDurationChange: (Long) -> Unit,
+    onComentsChange: (String) -> Unit
+) {
+  item {
+    val beginTimeTitle = "Effective time of start"
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/components/forms/BaseReportForm.kt:31]: _effective time of start of work_ 
would be clearer maybe

File: app/src/main/java/ch/epfl/skysync/models/flight/FinishedFlight.kt [hunk 4]
```diff
@@ -17,13 +18,21 @@ data class FinishedFlight(
...
+
+  private var flightStatus = FlightStatus.MISSING_REPORT
+
   override fun getFlightStatus(): FlightStatus {
-    return FlightStatus.COMPLETED
+    return this.flightStatus
+  }
+
+  fun reportCompleted() {
+    this.flightStatus = FlightStatus.COMPLETED
```
Comment (RAFInery) [inline app/src/main/java/ch/epfl/skysync/models/flight/FinishedFlight.kt:36]: make the class immutable to make sure the UI gets noticed of changes.

`getFlightStatus() `does check if 
**number of reports** == **number of team members**

adding a report creates a new FinishedFlight with the reportId added to the list of reports

**Review by RAFInery — APPROVED** [review]
LGTM. Please consider doing smaller PRs and adding some more information in the description (for UI some screen shots would be greate).

File: app/src/main/java/ch/epfl/skysync/components/TimePicker.kt [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+) {
+  when (displayMode) {
+    DisplayMode.Picker ->
+        IconButton(
+            modifier = modifier,
+            onClick = { onDisplayModeChange(DisplayMode.Input) },
+        ) {
+          Icon(
+              painter = painterResource(id = R.drawable.baseline_keyboard_24),
+              contentDescription = "",
```
[the remark this answers is not in the document]
Comment (Baribal02) [reply app/src/main/java/ch/epfl/skysync/components/TimePicker.kt:112]: Done

File: app/src/main/java/ch/epfl/skysync/components/forms/BaseReportForm.kt [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  item {
+    val endTimeTitle = "Effective time of end"
+    TimePickerField(defaultPadding, endTimeTitle, endTime) { onEndTimeChange(it) }
+  }
+
+  item { PauseField(defaultPadding, pauseDuration) { onPauseDurationChange(it) } }
+
+  item { VehicleProblemField(defaultPadding, flight.vehicles, onConfirm = onConfirm) }
+  items(vehicleProblem.keys.toList()) { vehicle ->
```
[the remark this answers is not in the document]
Comment (Baribal02) [reply app/src/main/java/ch/epfl/skysync/components/forms/BaseReportForm.kt]: Done

File: app/src/main/java/ch/epfl/skysync/components/TimePicker.kt [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+          )
+        }
+    DisplayMode.Input ->
+        IconButton(
+            modifier = modifier,
+            onClick = { onDisplayModeChange(DisplayMode.Picker) },
+        ) {
+          Icon(
+              painter = painterResource(id = R.drawable.baseline_schedule_24),
+              contentDescription = "",
```
[the remark this answers is not in the document]
Comment (Baribal02) [reply app/src/main/java/ch/epfl/skysync/components/TimePicker.kt:122]: Done

File: app/src/main/java/ch/epfl/skysync/components/forms/SpecialFields.kt [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        IconButton(
+            modifier = Modifier.padding(horizontal = defaultPadding).testTag("Add Problem Button"),
+            onClick = { showAddProblemDialog = true }) {
+              Icon(
+                  modifier = Modifier,
+                  imageVector = Icons.Default.Add,
+                  contentDescription = "Add Problem")
+            }
+      }
+  if (showAddProblemDialog) {
```
[the remark this answers is not in the document]
Comment (Baribal02) [reply app/src/main/java/ch/epfl/skysync/components/forms/SpecialFields.kt:125]: Done

File: app/src/main/java/ch/epfl/skysync/util/FormsUtil.kt (hunk 2 above)
Comment (Baribal02) [reply app/src/main/java/ch/epfl/skysync/util/FormsUtil.kt:10]: Done

**Pull request merged**